src/frontend/middleware.go

```diff
@@ -86,8 +87,13 @@ func ensureSessionID(next http.Handler) http.HandlerFunc {
 		var sessionID string
 		c, err := r.Cookie(cookieSessionID)
 		if err == http.ErrNoCookie {
-			u, _ := uuid.NewRandom()
-			sessionID = u.String()
+			if os.Getenv("ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS") == "" {
+				// Hard coded user id, shared across sessions
+				sessionID = "12345678-1234-1234-1234-123456789123"
+			} else {
+				u, _ := uuid.NewRandom()
+				sessionID = u.String()
+			}
```

Comment:
Nitpick (feel free to ignore):
We might be able to shorten this chunk of code.
```
sessionID = "12345678-1234-1234-1234-123456789123"
if os.Getenv("ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS") != "" {
	u, _ := uuid.NewRandom()
	sessionID = u.String()
}
```

Reply:
Yeah, I thought about something like this, but with this proposition we read that the default behavior/value is "a unique id across sessions" which is not the case. I think I would prefer the `if... else...` to clarify that point.

Reply:
**Question:** 
Is there a reason the default behaviour is having a single, shared session? Do you think it's more useful?
I think `ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS = true` should be the default behaviour because:
1. during live demos (or even on [onlineboutique.dev](https://onlineboutique.dev/)), each visitor should have their own cart (to avoid confusion)
2. it's what uses are already familiar with (since it's the _current_ behaviour)

**Question 2:**
In which, case should we use a different environment variable name such as `ENABLE_SINGLE_SHARED_SESSION` (where `ENABLE_SINGLE_SHARED_SESSION` means the opposite of `ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS`)?

Reply:
Yep, like I mentioned in the description:

> FIXME: before reviewing/merging, the test is with the value "" for a testing purpose, but it should be "true".

Sorry if it was not clear, for now with `""` it's just to test quickly, then I will set the proper value `"true"` in there ;)

Sorry for the confusion at this stage.

Reply:
RE: `ENABLE_SINGLE_SHARED_SESSION`, I'm not sure I get what you mean, could you elaborate?

Reply:
Ohhh, got it. What I understand: you'll be adding `ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS: true` to the `/kubernetes-manifests` (before merging this PR).
**Suggestion:** If that's the case, I suggest using an environment variable that means the opposite such as `ENABLE_SINGLE_SHARED_SESSION` (so that by default, we don't have to think about this variable at all).

Reply:
Sorry, nop, I'm having the test with `true` in the code directly, actually I just did that commit https://github.com/GoogleCloudPlatform/microservices-demo/pull/1023/commits/d236d8d736e2e27e31a4b9747cb2fcb1cfa8acde.

Does that make more sense now with this commit?

Reply:
Yup, just saw you commit. Thanks for that quick clarification!

I'll clarify what I mean by `ENABLE_SINGLE_SHARED_SESSION`:
* `ENABLE_SINGLE_SHARED_SESSION` means the opposite of `ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS`
* Right now, the default deployment of Online Boutique requires `ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS` to be set to `true`
* If, instead of `ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS`, we introduce `ENABLE_SINGLE_SHARED_SESSION` in this PR, we won't have to touch the `kubernetes-manifests/` and we could just introduce a new Kustomize Component similar [/kustomize/components/cymbal-branding](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/kustomize/components/cymbal-branding).
* Also, if we use `ENABLE_SINGLE_SHARED_SESSION`, the code could be:
```
u, _ := uuid.NewRandom()
sessionID = u.String()
if os.Getenv("ENABLE_SINGLE_SHARED_SESSION") != "" {
	sessionID = "12345678-1234-1234-1234-123456789123" // Hard code the single session shared by all visitors
}
```
or something similar.

But this is a minor detail, so I don't feel strongly.

Reply:
> Right now, the default deployment of Online Boutique requires ENABLE_UNIQUE_USER_ID_ACROSS_SESSIONS to be set to true

True, if you want the new behavior ("a unique user id accross sessions").

But not true, if you want to keep the default behavior ("new user id per session").

Reply:
Like discussed in 1:1, let's do `ENABLE_SINGLE_SHARED_SESSION`.